### ISSUE TITLE
undefined `this` when running ParseQuery with `.first`

### ISSUE THREAD

**Issue by ranby** [parse-community/Parse-SDK-JS#1613]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
When running a Parse.Query with `.first()`, I get an error like `TypeError: Cannot read properties of undefined (reading 'className') at eval (ParseQuery.js:1523:39)`

Parse.Query with .find works fine.
This seem to have broke in v3.5.0. Earlier versions work fine for me.

After digging into the source code a bit it looks like `this` in ParseQuery.js gets modified to `undefined` after QueryController has ran the async query request. 

### Steps to reproduce
Run a query like
```
const result = await new Parse.Query('Item').equalTo('objectId', '1234567890').first()
console.log({ result })
``` 

### Actual Outcome
A runtime error is thrown (see Logs section)

### Expected Outcome
Expected `result` would be an Parse.Object with the specified objectId or `undefined`

###  Environment

Client
- Parse JS SDK version: `3.5.0`

### Logs
Complete error:
```
TypeError: Cannot read properties of undefined (reading 'className')
    at eval (ParseQuery.js:1523:39)
```

**Comment by dblythy** [comment]
3.5.0 is broken, please see #1600. 

Can you try [3.5.1-alpha.2](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.1-alpha.2) or revert to 3.4.4

**Comment by ranby** [comment]
Aaah thank you @dblythy! 3.5.1-alpha.2 works! 

I would have saved a lot of time if I just had looked through closed issues as well 😅

**Comment by dblythy** [comment]
Closing as fixed via #1600
